open-apparel-registry/open-apparel-registry
# Enable contributors to remove individual facility list items from public display

## Overview

This PR enables list contributors to remove list items. Removing an item will keep it on a list but suppress it from displaying on the facility details page in any way. I designed this in line with the specs detailed in #594.

Connects #594 

## Demo

![Screen Shot 2019-06-20 at 5 37 38 PM](https://user-images.githubusercontent.com/4165523/59883144-23f80880-9382-11e9-865f-62511c4b6b93.png)

![Screen Shot 2019-06-20 at 5 37 46 PM](https://user-images.githubusercontent.com/4165523/59883145-23f80880-9382-11e9-904a-06b0a2ed8dca.png)

## Notes

In principle there is no reason why we cannot also remove list items with a status of "POTENTIAL_MATCH". However I did not have implement a UI for doing that, partly because the actions area for those rows is already kind of busy and partly because list items that are not yet matched don't show up on the facility details page anyway.

## Testing Instructions

- get branch then `./scripts/manage resetdb`, `./scripts/manage processfixtures`, `./scripts/server` and then sign in as c6@example
- in one tab, search for the facility named "Cotia". On its details page verify that there is a link to the contributor page for c6
- visit the facility lists for c6 and search for "Cotia". Use the UI to remove the facility and verify that the row changes its opacity on success and that you see the "REMOVED" status.
- try out the status filters and verify that "REMOVED" and the others work correctly
- look at the details page for "Cotia" and very that the contributor is no longer associated with the facility
- try removing some other facilities and verify that that also works. You may want to sign in as c2 or c8 and remove facilities from those lists, as you should test out that you can also remove "MATCHED" and "CONFIRMED_MATCH" list items -- and not just "NEW_FACILITY" items.


## Checklist
